status/status.go

```diff
@@ -255,22 +256,27 @@ func (s *Status) ReportSnapshotStatus(ctx context.Context, reporter ReporterInte
 			//integration test contains no changes
 			continue
 		}
-		testReport, err := s.generateTestReport(ctx, *integrationTestStatusDetail, snapshot)
-		if err != nil {
-			_ = WriteSnapshotReportStatus(ctx, s.client, snapshot, srs) // try to write what was already written
+		testReport, reportErr := s.generateTestReport(ctx, *integrationTestStatusDetail, snapshot)
+		if reportErr != nil {
+			if writeErr := WriteSnapshotReportStatus(ctx, s.client, snapshot, srs); writeErr != nil { // try to write what was already written
+				return fmt.Errorf("failed to generate test report AND write snapshot report status metadata: %w", errors.Join(reportErr, writeErr))
+			}
 			return fmt.Errorf("failed to generate test report: %w", err)
```

Comment:
this `err` here is only gods know what, but totally not error from `generateTestReport`

Reply:
sorry I'm blind :(
Updated now